### PR TITLE
Feat: sign protocol parameters in Genesis certificate

### PR DESCRIPTION
**Pull request by jpraynaud**
## Content
This PR includes an upgrade of the **Genesis certificate** to embed the protocol parameters of the next epoch.

## Pre-submit checklist

- Branch
  - [x] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

### CODE REVIEW

**Review by dlachaume — APPROVED** [review]
LGTM 👍

File: mithril-common/src/certificate_chain/certificate_genesis.rs [hunk 1]
```diff
@@ -41,6 +41,7 @@ impl CertificateGenesisProducer {
 
     /// Create the Genesis protocol message
     pub fn create_genesis_protocol_message(
+        genesis_protocol_paremeters: &ProtocolParameters,
```
[the remark this answers is not in the document]
Comment (jpraynaud) [reply mithril-common/src/certificate_chain/certificate_genesis.rs]: Done

**Review by sfauvel — APPROVED** [review]

File: mithril-common/src/entities/protocol_message.rs [hunk 2]
```diff
@@ -21,6 +21,12 @@ pub enum ProtocolMessagePartKey {
     #[serde(rename = "next_aggregate_verification_key")]
     NextAggregateVerificationKey,
 
+    /// The ProtocolMessage part key associated to the Next epoch protocol parameters
+    /// The protocol parameters that will be allowed to be used to sign during the next epoch
```
Comment (Alenar) [inline mithril-common/src/entities/protocol_message.rs:25]: Don't forget that this use a markdown like formatting ;).
```suggestion
    /// The ProtocolMessage part key associated to the Next epoch protocol parameters
    ///
    /// The protocol parameters that will be allowed to be used to sign during the next epoch
```

**Review by Alenar — APPROVED** [review]
LGTM :rocket:

File: mithril-common/src/entities/protocol_message.rs (hunk 2 above)
Comment (jpraynaud) [reply mithril-common/src/entities/protocol_message.rs:25]: Fixed :ok_hand:

**Pull request merged**